d2renderers/d2svg/table.go

```diff
@@ -113,18 +117,18 @@ func drawTable(writer io.Writer, targetShape d2target.Shape) {
 		tableHeader(headerBox, targetShape.Label, float64(targetShape.LabelWidth), float64(targetShape.LabelHeight), float64(targetShape.FontSize)),
 	)
 
-	fontSize := float64(targetShape.FontSize)
+	font := d2fonts.SourceSansPro.Font(targetShape.FontSize, d2fonts.FONT_STYLE_REGULAR)
 	var longestNameWidth float64
 	for _, f := range targetShape.SQLTable.Columns {
-		// TODO measure text
-		longestNameWidth = math.Max(longestNameWidth, float64(len(f.Name))*fontSize*5/9)
+		w, _ := ruler.MeasurePrecise(font, f.Name)
+		longestNameWidth = math.Max(longestNameWidth, w)
```

Comment:
yeah similar to what @ejulio-ts said. The right thing to do is get max width of any column and set the shape width to it, and then here each column is the shape width.

Reply:
this is just computing the offset for where to place the types, not for computing the total width of the table which we already have like you said